Content.Server/Chemistry/EntitySystems/SolutionRandomFillSystem.cs

```diff
@@ -0,0 +1,49 @@
+using Content.Server.Chemistry.Components.SolutionManager;
+using Content.Shared.Chemistry.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server.Chemistry.EntitySystems;
+
+public sealed class SolutionRandomFillSystem : EntitySystem
+{
+    [Dependency] private readonly SolutionContainerSystem _solutionsSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<RandomFillSolutionComponent, MapInitEvent>(OnRandomSolutionFillMapInit);
+    }
+
+    public void OnRandomSolutionFillMapInit(EntityUid uid, RandomFillSolutionComponent component, MapInitEvent args)
+    {
+
+        var target = _solutionsSystem.EnsureSolution(uid, component.Solution);
+        var sumOfWeights = 0;
+
+        foreach (var picked  in component.RandomList)
+        {
+            sumOfWeights += (int) picked.Weight;
+        }
```

Comment:
This is going to have rounding issues though looks like you picked this from elsewhere.

Realistically RandomSolutionEntry should implement an interface and then a helper method for this be added to randomextensions somewhere.